tests/qa_automation/patch_and_reboot.pm

```diff
@@ -33,7 +33,9 @@ sub run {
 
     pkcon_quit unless check_var('DESKTOP', 'textmode');
 
-    zypper_call(q{mr -d $(zypper lr | awk -F '|' '/NVIDIA/ {print $2}')}, exitcode => [0, 3]);
+    unless (script_run('zypper lr|grep -i nvidia')) {
+        zypper_call(q{mr -d $(zypper lr | awk -F '|' '{IGNORECASE=1} /nvidia/ {print $2}')});
```

Comment:
why `unless` thing? `exitcode [0, 3]` handles this situation and you need fewer calls

Reply:
zypper will return 0 when it will find nvidia repo, which is fail for `script_run`
If there are no nvidia repo then [this is not needed](https://openqa.suse.de/tests/3053874#step/patch_and_reboot/2)

Reply:
it works correctly ...:D returns 3 and this is allowed by exictcode :D

Reply:
removed my beautiful change, now we will still see this ugly failure ...